planner/core/plan.go

```diff
@@ -92,10 +92,44 @@ func optimizeByShuffle(pp PhysicalPlan, tsk task, ctx sessionctx.Context) task {
 		if shuffle := optimizeByShuffle4Window(p, ctx); shuffle != nil {
 			return shuffle.attach2Task(tsk)
 		}
+	case *PhysicalStreamAgg:
+		if shuffle := optimizeByShuffle4StreamAgg(p, ctx); shuffle != nil {
+			return shuffle.attach2Task(tsk)
+		}
 	}
 	return tsk
 }
 
+func optimizeByShuffle4StreamAgg(pp *PhysicalStreamAgg, ctx sessionctx.Context) *PhysicalShuffle {
+	concurrency := ctx.GetSessionVars().StreamAggConcurrency()
+	if concurrency <= 1 {
+		return nil
+	}
+
+	sort, ok := pp.Children()[0].(*PhysicalSort)
+	if !ok {
+		return nil
+	}
+
+	tail, dataSource := sort, sort.Children()[0]
+	reqProp := &property.PhysicalProperty{ExpectedCnt: math.MaxFloat64}
+	shuffle := PhysicalShuffle{
+		Concurrency:  concurrency,
+		Tail:         tail,
+		DataSource:   dataSource,
+		SplitterType: PartitionHashSplitterType,
+		ByItems:      pp.GroupByItems,
+	}.Init(ctx, pp.statsInfo(), pp.SelectBlockOffset(), reqProp)
+	return shuffle
+}
+
+func getPartitionSplitter(ctx sessionctx.Context) PartitionSplitterType {
+	if ctx.GetSessionVars().IsStreamAggRangePartitionEnabled() {
```

Comment:
I think we can remove this variable and always use `RangeSplitter` after it is implemented.